lifting-state-up/passing-down-event-handlers.md

```diff
@@ -0,0 +1,250 @@
+| `setStudentData(students);`                         | Ultimately, we want to update the `studentData` on state. We use our state update function, `setStudentData`, and we update it to our newly formed `students` array. |
+
+Our `App` now defines and holds a function that can update our `studentData`, named `updateStudentData`.
+
+## Send This Function to `StudentList`
+
+Now that we've defined `updateStudentData`, imagine if other components could _use_ this function.
+
+If other components could use this function, they'd have the ability to update the `studentData` in `App`! We want _the `Student` component_ to update the `studentData` in `App`.
+
+Other components _can_ use the `updateStudentData` function if they can reference it... and they can, _using props_.
+
+First, let's change `App` so that it sends this function to `StudentList` through a new prop named `onUpdateStudent`.
+
+<!-- prettier-ignore-start -->
+```js
+  return (
+    <main>
+      <h1>Attendance</h1>
+      <StudentList
+        students={studentData}
+        onUpdateStudent={updateStudentData}
+      ></StudentList>
+    </main>
+  );
+```
+<!-- prettier-ignore-end -->
+
+Let's update the PropTypes for `StudentList`. Our `StudentList` component now expects a `prop` named `onUpdateStudent`, whose value is a _function_.
+
+<!-- prettier-ignore-start -->
+```js
+StudentList.propTypes = {
+    students: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        nameData: PropTypes.string.isRequired,
+        emailData: PropTypes.string.isRequired,
+        isPresentData: PropTypes.bool
+    })),
+    onUpdateStudent: PropTypes.func
+};
+```
+<!-- prettier-ignore-end -->
+
+## Send This Function to `Student`
+
+Now, let's send this _exact same function_ from the `StudentList` component to the `Student` component.
+
+We access this function through `props`, and we send it to `Student` through `props`.
+
+The keys to reading and writing this code are to:
+
+1. Check that we're _reading_ the correct `prop` that was passed in
+2. Recognize that we can use any name for any `prop` we send to `Student`
+
+<!-- prettier-ignore-start -->
+```js
+const StudentList = (props) => {
+    const studentComponents = props.students.map((student, index) => {
+        return (
+            <li key={index}>
+                <Student
+                    id={student.id}
+                    name={student.nameData}
+                    email={student.emailData}
+                    isPresent={student.isPresentData}
+                    onUpdate={props.onUpdateStudent}
+                ></Student>
+            </li>
+        );
+    });
+
+    // ... return some JSX
+```
+<!-- prettier-ignore-end -->
+
+Here, we're saying that each `Student` component has a `prop` named `onUpdate`. The value of `onUpdate` is `props.onUpdateStudent`.
```

Comment:
say what we expect to be there (reinforcing we have to remember what these things are supposed to be)
```suggestion
Here, we're saying that each `Student` component has a `prop` named `onUpdate`. The value of `onUpdate` will be whatever was passed in `props.onUpdateStudent`, which we expect to be a reference to our `updateStudentData` function.
```